iRacing/iRacing.cs

```diff
@@ -608,50 +623,50 @@ public Data(DahlDesign dahlDesign)
-            Base.AddProp("ReactionTime", 0);
-            Base.AddProp("LEDWarnings", false);
+            Base.AttachDelegate("AnimationType", () => animaionType);
+            Base.AttachDelegate("ShiftLightRPM", () => shiftLightRPM);
+            Base.AttachDelegate("ReactionTime", () => Math.Round(reactionPush));
+            Base.AttachDelegate("LEDWarnings", () => LEDwarningActive);
 
-            Base.AddProp("LaunchBitePoint", 0);
-            Base.AddProp("LaunchSpin", 0);
-            Base.AddProp("LaunchIdealRangeStart", 0);
-            Base.AddProp("LaunchIdealRangeStop", 0);
-            Base.AddProp("LaunchGearRelease", 0);
-            Base.AddProp("LaunchGearReleased", 0);
-            Base.AddProp("LaunchTimeRelease", 0);
-            Base.AddProp("LaunchTimeReleased", 0);
-            Base.AddProp("HighPower", false);
-            Base.AddProp("LaunchThrottle", 0);
+            Base.AttachDelegate("LaunchBitePoint", () => clutchBitePoint);
+            Base.AttachDelegate("LaunchSpin", () => clutchSpin);
+            Base.AttachDelegate("LaunchIdealRangeStart", () => clutchIdealRangeStart);
+            Base.AttachDelegate("LaunchIdealRangeStop", () => clutchIdealRangeStop);
+            Base.AttachDelegate("LaunchGearRelease", () => clutchGearRelease);
+            Base.AttachDelegate("LaunchGearReleased", () => clutchGearReleased);
+            Base.AttachDelegate("LaunchTimeRelease", () => clutchTimeRelease);
+            Base.AttachDelegate("LaunchTimeReleased", () => clutchTimeReleased);
+            Base.AttachDelegate("HighPower", () => highPower);
+            Base.AttachDelegate("LaunchThrottle", () =>  launchThrottle);
 
 
             Base.AddProp("ApproximateCalculations", false);
             Base.AddProp("LapsRemaining", 0);
             Base.AddProp("LapBalance", 0);
 
-            Base.AddProp("LapStatus", 0);
+            Base.AttachDelegate("LapStatus", () => lapStatus);
 
             Base.AddProp("StintTimer", new TimeSpan(0));
             Base.AddProp("StintTotalTime", new TimeSpan(0));
             Base.AddProp("StintTotalHotlaps", 0);
             Base.AddProp("StintCurrentHotlap", 0);
-            Base.AddProp("StintValidLaps", 0);
-            Base.AddProp("StintInvalidLaps", 0);
+            Base.AttachDelegate("StintValidLaps", () => validStintLaps);
+            Base.AttachDelegate("StintInvalidLaps", () => invalidStintLaps);
 
             Base.AddProp("Pace", new TimeSpan(0));
 
-            Base.AddProp("PitBoxPosition", 1);
-            Base.AddProp("PitBoxApproach", false);
+            Base.AttachDelegate("PitBoxPosition", () => pitBox);  //TODO: check, was 1 before
+            Base.AttachDelegate("PitBoxApproach", () => boxApproach);
             Base.AddProp("PitEntry", false);
             Base.AddProp("PitSpeeding", false);
 
-            Base.AddProp("SessionBestLap", new TimeSpan(0));
+            Base.AttachDelegate("SessionBestLap", () => lapRecord);
```

Comment:
When setting new session best, it is set to lap record

Reply:
fixed